docs/coding.md

```diff
@@ -0,0 +1,78 @@
+# Coding Conventions
+
+Please follow coding conventions and guidelines described in the following documents:
+
+* [Go proverbs](https://go-proverbs.github.io/) - highly recommended read
+* [CodeReviewComments](https://github.com/golang/go/wiki/CodeReviewComments)
+* [Effective Go](https://golang.org/doc/effective_go.html)
+* [How to Write a Git Commit Message](https://chris.beams.io/posts/git-commit/)
+
+Here's a list of some more specific conventions that are often followed in
+the code and will be pointed out in the review process:
+
+## General
+
+* Keep variable names short for variables that are local to the function.
+* Do not export a function or variable name outside the package until you
+  have an external consumer for it.
+* Have setter or getter interfaces/methods to access/manipulate information in
+  a different package.
+* Do not use named return values in function definitions. Use only the type.
+  Exception: defer()'d functions.
+
+### Imports
+
+We use the following convention for specifying imports:
+
+```
+<import standard library packages>
+
+<import ceph-csi packages>
+
+<import third-party packages>
+```
```

Comment:
Is this really the convention here? I thought the proper convention was:
* stdlib
* 3rdparty
* local

If you look at the documenatation for "goimports" for example it says:
```
  -local string
        put imports beginning with this string after 3rd-party packages; comma-separated list
```

Reply:
Yes, corrected it.